src/Microsoft.ML.DataView/KeyDataViewType.cs

```diff
@@ -3,55 +3,73 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using Microsoft.ML.Internal.Utilities;
-using Microsoft.ML.Runtime;
 
 namespace Microsoft.ML.Data
 {
     /// <summary>
-    /// KeyTypes are for "id"-like data. The information happens to be stored in an unsigned integer
+    /// KeyDataViewTypes are for "id"-like data. The information happens to be stored in an unsigned integer
     /// type, but the information is not inherently numeric, so, typically, arithmetic is not
     /// meaningful. Examples are SSNs, phone numbers, auto-generated/incremented key values,
     /// class numbers, etc. For example, in multi-class classification, the label is typically
-    /// a class number which is naturally a KeyType.
+    /// a class number which is naturally a KeyDataViewType.
     ///
-    /// KeyTypes have a cardinality (i.e., <see cref="Count"/>) that is strictly positive.
+    /// KeyDataViewTypes have a cardinality (i.e., <see cref="Count"/>) that is strictly positive.
     ///
     /// Note that the underlying representation value does not necessarily match the logical value.
-    /// For example, if a KeyType has range 0-5000, then it has a <see cref="Count"/> of 5001, but
+    /// For example, if a KeyDataViewType has range 0-5000, then it has a <see cref="Count"/> of 5001, but
     /// the representational values are 1-5001. The representation value zero is reserved
     /// to mean a missing value (similar to NaN).
     /// </summary>
-    public sealed class KeyType : PrimitiveDataViewType
+    public sealed class KeyDataViewType : PrimitiveDataViewType
     {
-        public KeyType(Type type, ulong count)
+        public KeyDataViewType(Type type, ulong count)
             : base(type)
         {
-            Contracts.AssertValue(type);
-            if (count == 0 || type.ToMaxInt() < count)
-                throw Contracts.ExceptParam(nameof(count), "The cardinality of a {0} must not exceed {1}.MaxValue" +
-                    " and must be strictly positive but got {2}.", nameof(KeyType), type.Name, count);
+            if (!IsValidDataType(type))
+                throw new ArgumentException(
+                    $"Type is not valid, it must be {typeof(byte).FullName}, {typeof(ushort).FullName}, {typeof(uint).FullName}, or {typeof(ulong).FullName}.",
+                    nameof(type));
+            if (count == 0 || GetMaxInt(type) < count)
+                throw new ArgumentOutOfRangeException(
```

Comment:
>throw new ArgumentOutOfRangeException( [](start = 16, length = 38)

nit: you can use `Contracts.ExceptParam` since you moved it.

Reply:
Good call. Fixing.